### PR TITLE
Remove ProtoBuf encoding/decoding from RemoteClient

### PR DESCRIPTION
**Pull request by rkaippully**
`RemoteClient` decodes protobuf messages received from the gRPC service to a Haskell type to immediately encode it back to protobuf and send it to MQTT. This consumes a significant amount of memory and CPU.

This change removes the encoding/decoding logic completely from the `RemoteClient`. Instead, we pass the protobuf encoded bytes directly between the `Client` and the remote gRPC service. Any serialization errors will be handled by them instead of the `RemoteClient`.

### CODE REVIEW

File: src/Network/GRPC/MQTT/Wrapping.hs [hunk 1]
```diff
@@ -79,44 +65,6 @@ import Proto3.Wire.Decode (ParseError (..))
...
-    handler (ClientWriterRequest timeout metadata send)
-
-wrapBiDiStreamingClientHandler ::
-  (Message request, Message response) =>
-  (ClientRequest 'BiDiStreaming request response -> IO (ClientResult 'BiDiStreaming response)) ->
-  ClientHandler
-wrapBiDiStreamingClientHandler handler =
-  ClientBiDiStreamHandler \timeout metadata bidi -> do
-    handler (ClientBiDiRequest timeout metadata bidi)
-
```
Comment (Friede80) [inline src/Network/GRPC/MQTT/Wrapping.hs:119]: Very satisfying!

**Review by Friede80 — COMMENTED** [review]
This looks great! Should reduce a **lot** of overhead. 

Only issue is that when I run the tests I'm getting nondeterministic failures. I'm not sure if that's from changes made here or if there's a problem in `main`, but I think we should get that resolved before merging a significant change like this.

**Review by ixmatus — APPROVED** [review]

**Comment by ixmatus** [comment]
@Friede80's comment makes me realize that we should setup a GitHub Action to run the test suite for `grpc-mqtt`.

**Comment by rkaippully** [comment]
> This looks great! Should reduce a **lot** of overhead.
> 
> Only issue is that when I run the tests I'm getting nondeterministic failures. I'm not sure if that's from changes made here or if there's a problem in `main`, but I think we should get that resolved before merging a significant change like this.
> 
> Edit: the problem is present in `main`: #40

Somehow the tests passed for me when I ran them, but I can see random failures now. I'll defer merging till the cause is identified.

**Comment by rkaippully** [comment]
Tests are passing now with #41. Merging this.

**Pull request merged**